activemodel/test/cases/secure_password_test.rb

```diff
@@ -265,6 +265,11 @@ class SecurePasswordTest < ActiveModel::TestCase
     assert_equal false, @user.authenticate(" ")
   end
 
+  test "password_salt" do
+    @user.password = "secret"
+    assert_equal @user.password_digest.salt, @user.password_salt
+  end
```

Comment:
It wouldn't hurt to also have a `"password_salt returns nil when password is blank"` test here too, since `BCrypt::Password` would normally return a non-nil salt in that case:

```ruby
BCrypt::Password.create(" ").salt
# => "$2a$12$..."

BCrypt::Password.create(nil).salt
# => "$2a$12$..."
```

Reply:
Actually what happens here is... is it correct?

```ruby
password = ""
# password_salt -> nil

password = nil
# password_salt -> nil

password = " "
# password_salt -> "$2a$12$..."

```